File: 1250-longest-common-subsequence/1250-longest-common-subsequence.cpp
class Solution {
public:
    int longestCommonSubsequence(string text1, string text2) {
        vector<int> dp(text1.length(), 0);
        int longest = 0;

        for (char c : text2) {
            int currLen = 0;
            for (int i = 0; i < dp.size(); i++) {
                if (currLen < dp[i]) {
                    currLen = dp[i];
                } else if (c == text1[i]) {
                    dp[i] = currLen + 1;
                    longest = max(longest, currLen + 1);
                }
            }
        }

        return longest;
    }
};

File: 1250-longest-common-subsequence/1250-longest-common-subsequence.py
class Solution:
    def longestCommonSubsequence(self, text1: str, text2: str) -> int:
        dp = [0] * len(text1)

        longest = 0

        for c in text2:
            currLen = 0
            for i, val in enumerate(dp):
                if currLen < val:
                    currLen = val
                elif c==text1[i]:
                    dp[i] = currLen + 1
                    longest = max(currLen+1, longest)
                
        return longest